packages/core/saved-objects/core-saved-objects-server/src/model_version/transformations.ts

```diff
@@ -79,3 +79,15 @@ export interface SavedObjectModelBidirectionalTransformation<
    */
   down: SavedObjectModelTransformationFn<NewAttributes, PreviousAttributes>;
 }
+
+/**
+ * A data backfill function associated with a {@link SavedObjectsModelDataBackfillChange | data backfill} change.
+ *
+ * @remark Such transformation functions should only be used to backfill newly introduced fields.
+ *         Even if no check is performed to ensure that, using such transformations to mutate
+ *         existing data of the document can lead to data corruption or inconsistency.
+ */
+export type SavedObjectModelDataBackfillFn<
+  InputAttributes = unknown,
+  OutputAttributes = unknown
+> = SavedObjectModelTransformationFn<InputAttributes, OutputAttributes>;
```

Comment:
As discussed elsewhere, it could be desirable to more strictly control the kind of changes consumer code can make in a backfill by, for example, allowing them to set only a single field. One possible API:

```typescript
...
  changes: [
              {
                type: 'data_backfill',
                path: string[], // Maybe some clever typings here
                value: V,
              },
   ]
   // Or we could use dot path notation
...
```

Reply:
As discussed elsewhere, it could be desirable to more strictly control the kind of changes consumer code can make in a backfill by, for example, allowing them to set only a single field. One possible API:

```typescript
...
  changes: [
              {
                type: 'data_backfill',
                path: string[], // Maybe some clever typings here
                value: V,
              },
   ]
   // Or we could use dot path notation
...
```